src/components/PostDetails.tsx

```diff
@@ -66,29 +42,40 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
         postId: post.id,
       });
 
-      setComments(currentComments => [...currentComments, newComment]);
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
+      dispatch(addNewComment(newComment));
     } catch (error) {
-      // we show an error message in case of any error
-      setError(true);
+      dispatch(changeError());
     }
   };
 
   const deleteComment = async (commentId: number) => {
-    // we delete the comment immediately so as
-    // not to make the user wait long for the actual deletion
-    // eslint-disable-next-line max-len
-    setComments(currentComments =>
-      currentComments.filter(comment => comment.id !== commentId),
-    );
+    dispatch(removeComment(commentId));
```

Comment:
The 'removeComment' action is dispatched before the API call to delete the comment. This could lead to inconsistencies if the API call fails. Consider handling the API call success and failure cases appropriately.